src/tools/simeng/main.cc

```diff
@@ -58,233 +41,39 @@ int main(int argc, char** argv) {
-    //     // 0x321603E0, // orr w0, wzr, #1024
-    //     0x71000400,  // subs w0, w0, #1
-    //     // 0x320003E0, // orr w0, wzr, #1
-    //     // 0x71000400, // subs w0, w0, #1
-    //     0x54FFFFE1,  // b.ne -4
-    // };
-
-    // Out-of-order test; counts down from 1024*1024, with an independent `orr`
-    // at the start of each branch. With an instruction latency of 2 or greater,
-    // the `orr` at the start of the next loop should issue/execute while the
-    // preceding branch is waiting on the result from the `subs`.
-    uint32_t hex[] = {
-        // 0x321E03E0,  // orr w0, wzr, #4
-        // 0x321603E0,  // orr w0, wzr, #1024
-        0x320C03E0,  // orr w0, wzr, #1048576
-        0x320003E1,  // orr w0, wzr, #1
-        0x71000400,  // subs w0, w0, #1
-        // 0x00000000,  // invalid
-        0x54FFFFC1,  // b.ne -8
-                     // .exit:
-        0xD2800000,  // mov x0, #0
-        0xD2800BC8,  // mov x8, #94
-        0xD4000001,  // svc #0
-    };
-
-    // Load/store consistency test; a simple bubble sort algorithm
-    // uint32_t hex[] = {
-    //     0x320003E0,  //   orr w0, wzr, #1
-    //     0x51000400,  //   sub w0, w0, #1
-
-    //     0x11013001,  //   add w1, w0, #76
-    //                  // .start:
-    //     0x11000002,  //   add w2, w0, #0
-    //                  // .compare:
-    //     0xB9400044,  //   ldr w4, [x2, 0]
-    //     0xB9400445,  //   ldr w5, [x2, 4]
-    //     0x4B0400A6,  //   sub w6, w5, w4
-    //     0x37F80046,  //   tbnz w6, #31, #8 (.swap)
-    //     0x14000003,  //   b #12 (.next)
-    //                  // .swap:
-    //     0xB9000444,  //   str w4, [x2, 4]
-    //     0xB9000045,  //   str w5, [x2, 0]
-    //                  // .next:
-    //     0x11001042,  //   add w2, w2, #4
-    //     0x4B010046,  //   sub w6, w2, w1
-    //     0x37FFFEE6,  //   tbnz w6, #31, #-36 (.compare)
-    //     0x51001021,  //   sub w1, w1, #4
-    //     0x4B010006,  //   sub w6, w0, w1
-    //     0x37FFFE66,  //   tbnz w6, #31, #-52 (.start)
-    // };
-
-    // Force a load/store ordering violation
-    // uint32_t hex[] = {
-    //     0x320003E0,  //   orr w0, wzr, #1
-    //     0x51000400,  //   sub w0, w0, #1
-
-    //     0x11000002,  //   add w2, w0, #0
-    //     0x11013001,  //   add w1, w0, #76
-
-    //     0xB9000041,  //   str w1, [x2]
-    //     0xB9400043,  //   ldr w3, [x2]
-    //     0xB9000443,  //   str w3, [x2, 4]
-    // };
-
-    // Some arbitrary values to sort
-    // std::vector<int> memoryValues = {9,  6, 7, 20, 5,   0,  80, 2,  1,  6,
-    //                                  17, 4, 3, 22, 117, 11, 4,  12, 10, 18};
-    // memcpy(memory, memoryValues.data(), memoryValues.size() * sizeof(int));
-
-    process = std::make_unique<simeng::kernel::LinuxProcess>(
-        simeng::span<char>(reinterpret_cast<char*>(hex), sizeof(hex)), config);
+    coreInstance = std::make_unique<simeng::CoreInstance>(argc - 1, argv);
```

Comment:
Some comments about why `argc - 2` and `argv + 2` are performed may be useful, as it is not clear what is passed to each constructor